dbt_semantic_interfaces/protocols/saved_query.py

```diff
@@ -0,0 +1,39 @@
+from abc import abstractmethod
+from typing import Optional, Protocol, Sequence
+
+from dbt_semantic_interfaces.protocols.metadata import Metadata
+from dbt_semantic_interfaces.protocols.where_filter import WhereFilter
+
+
+class SavedQuery(Protocol):
+    """Represents a query that the user wants to run repeatedly."""
+
+    @property
+    @abstractmethod
+    def metadata(self) -> Optional[Metadata]:  # noqa: D
+        pass
+
+    @property
+    @abstractmethod
+    def name(self) -> str:  # noqa: D
+        pass
+
+    @property
+    @abstractmethod
+    def description(self) -> Optional[str]:  # noqa: D
+        pass
+
+    @property
+    @abstractmethod
+    def metric_names(self) -> Sequence[str]:  # noqa: D
+        pass
+
+    @property
+    @abstractmethod
+    def group_by_item_names(self) -> Sequence[str]:  # noqa: D
```

Comment:
I assume the idea is that `group_by_item_names` aren't required, but should be represented as an empty set when there are none? Was there a conversation somewhere about whether not we should mark them as optional (`Optional[Sequence[str]]`)? I'd just like to understand the logic so I can keep it in mind when reviewing other parts of the protocol and ensuring consistency.

Reply:
As a general rule, collections should not be typed as optional unless there's a meaningful difference between "no collection" and "empty collection".

So we should not make these optional return types, but on implementation we can default them to an empty Sequence type so users are not required to pass in a value.

Reply:
Perfect, that works for me 🙂